OpenRA.Mods.Common/Scripting/Properties/DeliveryProperties.cs

```diff
@@ -21,33 +21,40 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("Ability")]
 	public class DeliversCashProperties : ScriptActorProperties, Requires<IMoveInfo>, Requires<DeliversCashInfo>
 	{
-		readonly DeliversCashInfo info;
+		readonly DeliversCash deliversCash;
 
 		public DeliversCashProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			info = Self.Info.TraitInfo<DeliversCashInfo>();
+			deliversCash = Self.Trait<DeliversCash>();
 		}
 
 		[ScriptActorPropertyActivity]
 		[Desc("Deliver cash to the target actor.")]
 		public void DeliverCash(Actor target)
 		{
+			var acceptsDeliveredCash = target.TraitOrDefault<AcceptsDeliveredCash>();
+			if (acceptsDeliveredCash == null)
+				throw new LuaException("Actor '{0}' does not accept delivered cash!".F(target));
+
+			if (acceptsDeliveredCash.IsTraitDisabled || deliversCash.IsTraitDisabled)
```

Comment:
The trait being disabled should be treated the same way as the trait not being defined - but i'm not sure which of silently returning or throwing an exception is the best result.

Reply:
I don't think there is a reliable way to check if the trait is disabled in Lua, so i think the throwing an exception here could cause issues.

Reply:
This hasn't been a problem so far for other conditional traits (e.g. capturing, attacking) so far. Once it does become a problem we will need to fix this everywhere in one go.